File: dotnet/Atlas.Averbacao/Api/Contrato/ContratoController.cs
using Atlas.Averbacao.Api.Contratacao;
using Atlas.Averbacao.Api.Contrato;
using Atlas.Averbacao.Services.Contrato;
using Microsoft.AspNetCore.Mvc;

namespace Atlas.Averbacao.Api.Contrato;

[ApiController]
[Route("v1/contratos/averbados")]
public sealed class ContratoController(IContratoService contrato) : ControllerBase
{
    /// <summary>O contrato foi averbado na folha.</summary>
    [HttpPost]
    [ProducesResponseType(typeof(OkResponse), 200)]
    public async Task<OkResponse> Post([FromBody] ContratoAverbadoRequest request)
    {
        await contrato.Averbado(request);
        return new OkResponse();
    }
}

File: java/src/main/java/br/com/atlas/averbacao/api/contrato/ContratoController.java
package br.com.atlas.averbacao.api.contrato;

import br.com.atlas.averbacao.api.contratacao.OkResponse;
import br.com.atlas.averbacao.services.contrato.ContratoContract;
import io.swagger.v3.oas.annotations.Operation;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ContratoController {
    private final ContratoContract contrato;

    public ContratoController(ContratoContract contrato) {
        this.contrato = contrato;
    }

    @Operation(summary = "O contrato foi averbado na folha")
    @PostMapping("/v1/contratos/averbados")
    public OkResponse post(@Valid @RequestBody ContratoAverbadoRequest request) {
        contrato.averbado(request);
        return new OkResponse();
    }
}
